c10/util/TypeIndex.h

```diff
@@ -93,17 +93,16 @@ inline constexpr uint64_t type_index_impl() {
 
 template <typename T>
 inline constexpr type_index get_type_index() {
-#if !defined(__CUDA_ARCH__)
+#if defined(__CUDA_ARCH__)
+  // Don't run this on device code
+  static_assert(false && sizeof(T));
+  return type_index(0);
+#endif
```

Comment:
Why not simplify it to something like the following? (Or, if it still needs ifdef, why return is required there?
```suggestion
  static_assert(!defined(__CUDA_ARCH__) && sizeof(T), " Don't call me from device code");
```